### ISSUE TITLE
Nock doesn't work with v1.3

### ISSUE THREAD

**Issue by JustinBeckwith** [olalonde/follow-redirects#77]
Greetings!  I'm using this library indirectly, through the `axios` npm module.  When 1.3 was released, all of the sudden my CI for node.js 4 and 6 (which don't have a package-lock.json) started to fail randomly.  After a log of digging, I discovered that this commit:
https://github.com/olalonde/follow-redirects/commit/847a47fb00ea2dcb7a8886ca16d9d82dc6accb5d

Before that commit - I could use axios and nock together in my tests.  After that commit, it causes bizarre behavior where requests just seem to hang or immediately exit.  Here's some code to reproduce the issue:

```js
const axios = require('axios');
const nock = require('nock');
const baseUrl = 'https://accounts.google.com:443';
const path = '/o/oauth2/token';
nock(baseUrl).post(path).reply(200);
axios({
  url: baseUrl + path,
  method: 'POST',
  data: { 'test': true }
}).then(r => {
  console.log("STATUS: " + r.status);
}).catch(e => {
  console.error("ERROR: " + e.message);
});
```

This code will work as expected in 1.2.6, but fails to work with 1.3.0.  It also works fine if I leave nock out of the equation, so I believe at this point it's some interaction with nock and that commit of your library that's causing our issues.  

I'm going to keep digging, but I wanted to get this thread started.

**Comment by JustinBeckwith** [comment]
Also as a note - this bug very specifically only happens when using a POST with data.

**Comment by RubenVerborgh** [comment]
Thanks for catching this so early on!

**Comment by RubenVerborgh** [comment]
It's a Nock problem, reported in node-nock/nock#1046 and fixed in node-nock/nock#1047.

**Comment by JustinBeckwith** [comment]
Awesome.  Thank you for digging into this!
